src/OrleansAzureUtils/Providers/Storage/AzureBlobStorage.cs

```diff
@@ -189,33 +189,30 @@ private static string GetBlobName(string grainType, GrainReference grainId)
                 var blob = container.GetBlockBlobReference(blobName);
                 blob.Properties.ContentType = "application/json";
 
-                var containerNotFound = false;
                 try
                 {
-                    await blob.UploadTextAsync(
-                            json,
-                            Encoding.UTF8,
-                            AccessCondition.GenerateIfMatchCondition(grainState.ETag),
-                            null,
-                            null).ConfigureAwait(false);
+                    await DoConditionalUpdate(() => blob.UploadTextAsync(json, Encoding.UTF8, AccessCondition.GenerateIfMatchCondition(grainState.ETag), null, null),
+                        blobName, grainState.ETag).ConfigureAwait(false);
                 }
                 catch (StorageException exception)
                 {
+                    if (exception.IsPreconditionFailed())
```

Comment:
Won't this always be false given it would otherwise be caught in `DoConditionalUpdate`?